Fix help sections for CLI options

The-Compiler opened the pull request:
Probably best reviewed commit by commit. Notably, this fixes:

    pytest-warnings:
      -W, --pythonwarnings PYTHONWARNINGS
                            Set which warnings to report, see -W option of Python itself
      --maxfail=num         Exit after first num failures or errors
      --strict-config       Any warnings encountered while parsing the `pytest` section of the configuration file raise errors
      --strict-markers      Markers not registered in the `markers` section of the configuration file raise errors
      --strict              (Deprecated) alias to --strict-markers
      -c, --config-file FILE
                            Load configuration from `FILE` instead of trying to locate one of the implicit configuration files.
      --continue-on-collection-errors
                            Force test execution even if collection errors occur
      --rootdir=ROOTDIR     Define root directory for tests. Can be relative path: 'root_dir', './root_dir', 'root_dir/another_dir/'; absolute path: '/home/user/root_dir'; path with variables: '$HOME/root_dir'.

which makes no sense for anything except the first option, and is a regression in 19e99ab4131fbd709ce0e5a17694e1be3b22f355.

Other than that, it moves:

- `--continue-on-collection-errors` to "collection" rather than "general"
- `--config-file` and `--rootdir` to "test session [...] configuration" (where we also have things like `--basetemp` and `--override-ini`) instead of "general"

Finally, it cleans up incorrectly cargo-culted usage of `group._addoption` instead of `group.addoption`.

Not sure if this needs a changelog?

nicoddemus reviewed (approved):
Looks great, thanks for the cleanup!

I think a "trivial" changelog is good to have, given this is a minor but still user facing change.

The-Compiler commented:
Added changelog and rebased (no other changes). We don't have a `trivial` changelog category anymore, so I went for `doc` over `misc`.

Any opinions on whether to backport this (IMHO yes), and whether to squash-merge (IMHO no)?

nicoddemus commented:
> Any opinions on whether to backport this (IMHO yes)

IMHO no, I really prefer to stick to docs-only and bugs-only for patch releases -- although this could be considered "doc", is still output from running pytest, who knows who might be parsing this and possibly breaking due to it changing? But I understand this is really low risk, so is not a strong stance, so please go ahead if you would like to see this in the next patch release.

> and whether to squash-merge (IMHO no)?

Agreed!

The-Compiler commented:
Whoops, I agree with what you said but forgot I already put the backport label on. Let's just leave it un-backported, given that nobody noticed the bugfix part for almost 4.5 years, I'm not in a hurry. 😅
